credoai/reporting/dataset_base.py

```diff
@@ -0,0 +1,186 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+
+from credoai.reporting.credo_report import CredoReport
+from credoai.reporting import plot_utils
+
+
+class DatasetModuleReport(CredoReport):
+    def __init__(self, module):
+        super().__init__()
+        self.module = module
+
+    def create_report(self, filename=None):
+        """Creates a fairness report for dataset assessment module
+
+        Parameters
+        ----------
+        filename : string, optional
+            If given, the location where the generated pdf report will be saved, by default Non
+
+        Returns
+        -------
+        array of figures
+        """
+        results_all = self.module.get_results()
+
+        # Generate sample balance barplots
+        fig, axs = plt.subplots(nrows=3, figsize=(8, 8), dpi=150)
+        results = results_all["balance_metrics"]["sample_balance"]
+        df = pd.DataFrame(results)
+        sensitive_feature_name = list(df.drop(["count", "percentage"], axis=1).columns)[
+            0
+        ]
+
+        ax = sns.barplot(
+            x="count",
+            y=sensitive_feature_name,
+            data=df,
+            palette=plot_utils.credo_diverging_palette(1),
+            alpha=1,
+            ax=axs[0],
+        )
+        fig.patch.set_facecolor("white")
+        sns.despine()
```

Comment:
I changed all `frame_off` to `despine`, which keeps the left and bottom axes' lines. Think it looks better